pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -390,6 +399,32 @@ class Timestamp(_Timestamp):
             # User passed tzinfo instead of tz; avoid silently ignoring
             tz, tzinfo = tzinfo, None
 
+        # Allow fold only for unambiguous input
+        if fold is not None:
+            if fold not in [0, 1]:
+                raise ValueError(
```

Comment:
Could you add a test that hits this exception?

Reply:
Done.